test/test_transforms.py

```diff
@@ -987,19 +987,23 @@ def test_2d_ndarray_to_pil_image(self):
                 self.assertTrue(np.allclose(img_data, img))
 
     def test_tensor_bad_types_to_pil_image(self):
-        with self.assertRaises(ValueError):
+        with self.assertRaisesRegex(ValueError, r'pic should be 2/3 dimensional. Got \d+ dimensions.'):
             transforms.ToPILImage()(torch.ones(1, 3, 4, 4))
+        with self.assertRaisesRegex(ValueError, r'pic should not have > 4 channels. Got \d+ channels.'):
+            transforms.ToPILImage()(torch.ones(6, 4, 4))
 
     def test_ndarray_bad_types_to_pil_image(self):
         trans = transforms.ToPILImage()
-        with self.assertRaises(TypeError):
+        with self.assertRaisesRegex(TypeError, r'Input type \w+ is not supported'):
```

Comment:
Could you please also fix this test as it was also incorrect: add `with self.assertRaisesRegex(TypeError, r'Input type \w+ is not supported')` for each `trans(np.ones([4, 4, 1], type))`.